numba_mpi/api/requests.py

```diff
@@ -0,0 +1,199 @@
+""" Utilities for handling MPI_Requests with respective Wait and Test function
+    wrappers.
+"""
+import ctypes
+
+import numba
+import numpy as np
+
+from numba_mpi.common import (
+    _MpiRequestNpType,
+    _MpiRequestPtr,
+    _MpiStatusPtr,
+    create_status_buffer,
+    libmpi,
+)
+
+# helper function to allocate numpy array of request handles
+
+
+@numba.njit
+def create_requests_array(count=1):
+    """Helper function for creating numpy array storing pointers to MPI_Request handles."""
+    return np.empty(count, dtype=_MpiRequestNpType)
```

Comment:
I'd suggest exposing something lik `_MpiRequestNpType` as a part of the public API, and then leave it to the user if the requests pointers would be stored in np.arrays, tuples or other containers

Reply:
done